src/commands/infra/generate.ts

```diff
@@ -175,3 +175,95 @@ export const validateRemoteSource = async (
   }
   return true;
 };
+
+/**
+ * Creates "generated" directory if it does not already exists
+ *
+ * @param projectPath Path to the definition.json file
+ */
+export const createGenerated = async (projectPath: string) => {
+  const newGeneratedPath = projectPath + "-generated";
+  mkdirp(newGeneratedPath, err => {
+    if (err) {
+      logger.error(`An problem occured when creating a generated direcotry`);
+      logger.error(err);
+    }
+    logger.info(`Created generated directory: ${newGeneratedPath}`);
+  });
+  return newGeneratedPath;
+};
+
+/**
+ * Parses the definition.json file and copies the appropriate template
+ * to "-generated" directory
+ *
+ * @param projectPath Path to the definition.json file
+ * @param generatedPath Path to the generated directory
+ */
+export const parseDefinitionJson = async (projectPath: string) => {
+  const definitionJSON = await readDefinitionJson(projectPath);
+  const source = definitionJSON.source;
+  const httpReg = /^(.*?)\.com/;
+  const punctuationReg = /[^\w\s]/g;
+  let sourceFolder = source.replace(httpReg, "");
+  sourceFolder = sourceFolder.replace(punctuationReg, "_").toLowerCase();
+  const templatePath = path.join(
+    spkTemplatesPath,
+    sourceFolder,
+    definitionJSON.template
+  );
+  return templatePath;
+};
+
+/**
+ * Creates a spk.tfvars in "-generated" directory
+ *
+ * @param projectPath Path to the definition.json file
+ * @param generatedPath Path to the generated directory
+ */
+export const generateSpkTfvars = async (
+  projectPath: string,
+  generatedPath: string
+) => {
+  try {
+    // Remove existing spk.tfvars if it already exists
+    if (fs.existsSync(path.join(generatedPath, "spk.tfvars"))) {
+      fs.unlinkSync(path.join(generatedPath, "spk.tfvars"));
+    }
+    // Parse definition.json and extract "variables"
+    const definitionJSON = await readDefinitionJson(projectPath);
+    const variables = definitionJSON.variables;
+    // Restructure the format of variables text
```

Comment:
Consider adding a little more description about the transform you are making here in the comments.  As someone approaching the project with not a lot of context, its hard to understand what all of the regexes are doing, and why they need to do what they are doing.  :)